### PR TITLE
👌 IMPROVE: minor adjustments to the running workflows module

### LINKED ISSUE #356: 👌 IMPROVE: "Running workflows" module
After a discussion with @ramirezfranciscof and @flavianojs (also see https://github.com/aiidateam/aiida-tutorials/issues/339#issuecomment-854871294), we decided that the second hands-on session should focus on running calculations in general, i.e. not just running _calculation jobs_. This was a very good point raised by @ramirezfranciscof, in that it would be weird to split up the example for running a single calculation and then running a full workflow, with the data section in between.

So, we'll move the `PwBandsWorkChain` related content from the BIGMAP/ABC/QEschool into a separate module on "Running workflows". This will show the user:

* How to see what workflows are available (`verdi plugin list`) and load the work chain
* How to easily set up a pre-populated builder with the `get_builder_from_protocol()` method.
* How to run and monitor a work chain

This makes for a rather small module though. While this is not necessarily a big problem, one option of making this somewhat more interesting is having them import a bunch more structures and loop over them to submit several work chains. It will become clear that you have to start organising your data somewhat in case you do this, and so this naturally guides them to the concept of groups. Just an idea!

### PR DESCRIPTION
**Pull request by ramirezfranciscof**
Removed a couple of mentions to other modules/sections, re-ordered a bit and added a couple of titles for clearer divide.

An "important" feature I added at the end is a summary of the most important lessons of the module. I did it as an admonition because I didn't want it showing in the sidebar as a title, but I'm not sure this is the best way to go. At least with the current format it doesn't look super nice: maybe we could change the CSS or design a different kind of box for this?

fix #356

### CODE REVIEW

File: docs/sections/calculations/workflows.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 # Running workflows
 
-AiiDA can help you run individual calculations, but it is really designed to help you run workflows that involve several calculations, while automatically keeping track of the provenance for full reproducibility.
+AiiDA can help you run individual calculations, but it is really designed to help you run workflows that involve several of them, while automatically keeping track of the provenance for full reproducibility.
```
Comment (CasperWA) [inline docs/sections/calculations/workflows.md]: I personally prefer the explicitness of the original sentence here...

File: docs/sections/calculations/workflows.md [hunk 2]
```diff
@@ -21,7 +21,10 @@ This is a fully automated workflow that will:
 3. Run a self-consistent field calculation on the refined structure.
 4. Run a band structure calculation at a fixed Kohn-Sham potential along a standard path between high-symmetry k-points determined by [SeeK-path][seek-path].
 
-In order to run it, we will again open the `verdi shell`.
+
+## Submitting a work chain
+
+In order to run this work chain, we will again open the `verdi shell`.
```
Comment (CasperWA) [inline docs/sections/calculations/workflows.md]:
```suggestion
In order to run this work chain, you should again open the `verdi shell`.
```

File: docs/sections/calculations/workflows.md [hunk 3]
```diff
@@ -21,7 +21,10 @@ This is a fully automated workflow that will:
...
 We will then load the work chain using its entry point and the `WorkflowFactory`:
```
Comment (CasperWA) [inline docs/sections/calculations/workflows.md]:
```suggestion
You should then load the work chain using its entry point and the `WorkflowFactory`:
```

File: docs/sections/calculations/workflows.md [hunk 4]
```diff
@@ -34,14 +37,17 @@ Setting up the inputs one by one as we did for the pw.x calculation in the previ
 Instead, we are going to use one of the protocols that has been set up for the workflow.
 To do this, all we need to provide is the code and initial structure we are going to run:
 
-```{code-block}
+:::{margin}
+Replace `<CODE_PK>` and `<STRUCTURE_PK>` with those of the code and structure you used in the base module of this section.
```
Comment (CasperWA) [inline docs/sections/calculations/workflows.md]:
```suggestion
Replace `<CODE_PK>` and `<STRUCTURE_PK>` with the respective PKs of the code and structure you used in the base module of this section.
```

File: docs/sections/calculations/workflows.md [hunk 5]
```diff
@@ -131,5 +119,32 @@ PwBandsWorkChain<113> Finished [0] [7:results]
 The bracket `[7:result]` indicates the current step in the outline of the `PwBandsWorkChain` (step 7, with name `result`).
 The `process status` is particularly useful for debugging complex work chains, since it helps pinpoint where a problem occurred.
 
-Congratulations on finishing the first part of the tutorial!
-In the next section, we'll look at how to organize and query your data.
+
+## Displaying the results
+
+Once the work chain has finished running, use `verdi process show <PK>` to inspect the `PwBandsWorkChain` and find the PK of its `band_structure` output.
+We can also plot the band structure using the `verdi shell`:
```
Comment (CasperWA) [inline docs/sections/calculations/workflows.md]:
```suggestion
You can also plot the band structure using the `verdi shell`:
```

File: docs/sections/calculations/workflows.md [hunk 6]
```diff
@@ -131,5 +119,32 @@ PwBandsWorkChain<113> Finished [0] [7:results]
...
+It should look similar to the one shown here:
+
+:::{figure} include/images/si_bands.png
+:width: 100%
+
+Band structure computed by the `PwBandsWorkChain`.
+
+:::
+
+:::{important} **What we learnt**
```
Comment (CasperWA) [inline docs/sections/calculations/workflows.md]: Or maybe something less demeaning? Like "What this tutorial teaches/has teached you" or "Summary of key points in this section/tutorial"?
```suggestion
:::{important} **What you learnt**
```

File: docs/sections/calculations/workflows.md [hunk 7]
```diff
@@ -131,5 +119,32 @@ PwBandsWorkChain<113> Finished [0] [7:results]
...
+
+ - Work chains can be prepared from a builder and submitted just like calculations.
```
Comment (CasperWA) [inline docs/sections/calculations/workflows.md]:
```suggestion
 - Work chains can be prepared from a builder and submitted similar to calculations.
```

File: docs/sections/calculations/workflows.md [hunk 8]
```diff
@@ -131,5 +119,32 @@ PwBandsWorkChain<113> Finished [0] [7:results]
...
+ - The method `get_builder_from_protocol()` can return a pre-populated builder from minimal inputs.
+ - In addition to `verdi process list`, you can see a hierarchical overview of a workflow by using `verdi process status`
```
Comment (CasperWA) [inline docs/sections/calculations/workflows.md]:
```suggestion
 - In addition to `verdi process list`, you can see a hierarchical overview of a workflow by using `verdi process status`.
```

**Review by CasperWA — DISMISSED** [review]
Some things:

- `workchain` versus `work chain`.
- You/We (#360)

File: docs/sections/calculations/workflows.md (hunk 6 above)
Comment (ramirezfranciscof) [reply docs/sections/calculations/workflows.md]: Haha demeaning? I agree one needs to be careful with the general tone but I don't think people will get offended by a title like that.

Anyways, what do you think of

```suggestion
:::{important} **Key takeaways**
```

Comment (CasperWA) [reply docs/sections/calculations/workflows.md]: I like it 👍 But maybe lose the capitalization of `Takeaways`?

File: docs/sections/calculations/workflows.md (hunk 1 above)
Comment (ramirezfranciscof) [reply docs/sections/calculations/workflows.md]: I went back to being explicit but used "sub-processes" to be less repetitive.

**Comment by ramirezfranciscof** [comment]
@mbercx :

1. What do we do about "workchains" vs "work chains". I personally like "workchains" better but I think you were using "work chains" in the workflow section, no?
2. Do you have some feedback about how to better display the "Key takeaways" at the end?

**Comment by mbercx** [comment]
> 1. What do we do about "workchains" vs "work chains". I personally like "workchains" better but I think you were using "work chains" in the workflow section, no?

Why do you like "workchains" better? Would you also use workfunction? I've always preferred using work function and work chain consistently.

EDIT: This may be due to my aversion to the Dutch language. We tend to make [ridiculously long closed compound words](https://www.mosalingua.com/en/longest-word-in-the-world/#longest-dutch-word-meervoudigepersoonlijkheidsstoornissen). In this [nice blog on compound words](https://www.grammarly.com/blog/open-and-closed-compound-words/), the difference between open (work chain) and closed (workchain) compound words is discussed. Although "workflows" is now a "real word", I'm not sure if "workchain" is, so I would write it as an open compound word.

> 2. Do you have some feedback about how to better display the "Key takeaways" at the end?

Would probably be nicer to have a "Key takeaways" directive, so you don't have to add both "Important" and "Key takeaways". An alternative would be to use a `sphinx-panel`. 

EDIT: Probably best to use panels for this. For more complicated modules (I'm looking at you Writing work chains basics module) this will allow us to give more structure to the "Key takeaways". No need to do this here though, happy to work on this as well, or discuss how we should uniformalize this.

**Comment by ramirezfranciscof** [comment]
> Why do you like "workchains" better? Would you also use workfunction? I've always preferred using work function and work chain consistently.
> 
> EDIT: This may be due to my aversion to the Dutch language. We tend to make [ridiculously long closed compound words](https://www.mosalingua.com/en/longest-word-in-the-world/#longest-dutch-word-meervoudigepersoonlijkheidsstoornissen). In this [nice blog on compound words](https://www.grammarly.com/blog/open-and-closed-compound-words/), the difference between open (work chain) and closed (workchain) compound words is discussed. Although "workflows" is now a "real word", I'm not sure if "workchain" is, so I would write it as an open compound word.

Actually, yes =P I think I prefer workchains, workfunctions, calcfunctions and calcjobs to all be single words. I would definitively not generalize this to all things like you mention in Dutch and I also know of German, but it is actually to emphasize that a workchain is something much more specific than "a chain of work".
 
> Would probably be nicer to have a "Key takeaways" directive, so you don't have to add both "Important" and "Key takeaways". An alternative would be to use a `sphinx-panel`.
> 
> EDIT: Probably best to use panels for this. For more complicated modules (I'm looking at you Writing work chains basics module) this will allow us to give more structure to the "Key takeaways". No need to do this here though, happy to work on this as well, or discuss how we should uniformalize this.

I like the idea of the panel as well. I could try to take a look and see if I can do a first draft and you can modify it later, where is the format for panels we have in the indexes? Do I put this in the same place?

**Comment by mbercx** [comment]
> Actually, yes =P I think I prefer workchains, workfunctions, calcfunctions and calcjobs to all be single words. I would definitively not generalize this to all things like you mention in Dutch and I also know of German, but it is actually to emphasize that a workchain is something much more specific than "a chain of work".

Haha, then we disagree on this. I prefer non-compounded versions (especially e.g. calcfunction -> calculation function). Maybe we should have a vote on this then. Although I'm also happy being the tutorial dictator. 😎 

> I like the idea of the panel as well. I could try to take a look and see if I can do a first draft and you can modify it later, where is the format for panels we have in the indexes? Do I put this in the same place?

Format? The format isn't really described anywhere, but you can adapt it from the indexes. You'll have to adapt the syntax to MyST though:

https://sphinx-panels.readthedocs.io/en/latest/

**Comment by mbercx** [comment]
@ramirezfranciscof I'll open an issue for the compound word issue, so don't worry about it in this PR.

EDIT: see https://github.com/aiidateam/aiida-tutorials/issues/377

**Review by mbercx — APPROVED** [review]
Merge first, think later! 🚀

**Pull request merged**